system/gatekeeper/templates/constrainttemplate-region-value-mismatch.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: gkregionvaluemismatch
+spec:
+  crd:
+    spec:
+      names:
+        kind: GkRegionValueMismatch
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          type: object
+          properties:
+            helmManifestParserURL:
+              type: string
+            region:
+              type: string
+
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package gkregionvaluemissing
+
+        helm_release := input.review.object
+        parser_url := sprintf("%s/v3", [input.parameters.helmManifestParserURL])
+        parser_resp := http.send({"url": parser_url, "method": "POST", "raise_error": false, "raw_body": helm_release.data.release, "timeout": "15s"})
+
+        violation[{"msg": msg}] {
+          helm_release.kind == "Secret"
+          helm_release.type == "helm.sh/release.v1"
+          helm_release.metadata.labels.status == "deployed"
+
+          parser_resp.status_code == 200
+          input.parameters.region != parser_resp.body.values.global.region
```

Comment:
This check should only trigger when `.Values.global.region` contains a valid value, i.e. when `GkRegionValueMissing` does not trigger. In other words, you should use `object.get()` to traverse through the values and check if you're getting a string value, similar to how `GkRegionValueMissing` is implemented now.

Please add a testcase where `.Values.global.region` is not set at all, and also one where it is set to an invalid value like `null`.

Reply:
This was marked as resolved, but has not been resolved. Since we now have the lib module setup, it may be useful to move `is_region_name()` from the other policy into a lib module to reuse this code here.